### PR TITLE
added IBL Sheen support

### LINKED ISSUE #22359: Sheen not working in combination with hdr env?
**Describe the bug**
Sheen rendering doesn't seem to work in combination with an HDR environment map.

**To Reproduce**

1. Load an HDR environment map
2. Set the scene's *environment* property to that map
3. Add some generic mesh (e.g. sphere) to the scene
4. Create a MeshPhysicsMaterial and set its sheen property to some color value.
5. Assign the material to the mesh.

**Expected behavior**

The sheen should render even when using an HDR environment map, Instead, you have to set the *envMapIntensity* property of the material to a very small value or zero to be able to see the sheen at all.

**Platform:**

 - Device: Desktop
 - OS: Windows
 - Browser: Chrome
 - Three.js version: 131

### PR DESCRIPTION
**Pull request by elalish**
Fixes #22359 

@WestLangley @mrdoob 

I finally did the math to  curve fit the integrated BRDF for sheen so it can be  applied to the IBL (filament and gltf sample viewer use a LUT stored in a texture instead). That approximation is  pretty good. The really wild approximation is using our PMREM for the sheen lookup, since the sheen BRDF is very different from Trowbridge-Reitz (our base roughness model) and it's anisotropic. All the realtime renderers  I've seen do this, but they sample their PMREM in different ways. Some choose to sample at the reflection vector and the sheen roughness value, but I don't think this is a very good choice (not that anything is very  accurate). I opted for using our diffuse sample instead (the sheen BRDF is close to a tangent-plane circle for all roughness values, so this at least pulls light evenly from this circle, though also from above), and has the major advantage of not requiring an extra texture lookup. My way also ensures you don't get an image reflection when sheen roughness goes to zero, which may happen with the other approach. 

I did my  best to follow the  style of the ClearCoat extension. I have an energy preservation term in  there too that also came from my curve fitting. The sheen layer  is  the top layer, above clear coat, which  seems reasonable, as it might represent a dusty surface. 

Here are some example renders:
model-viewer-golden
model-viewer-golden

### CODE REVIEW

**Comment by WestLangley** [comment]
I will have a look...

**Comment by elalish** [comment]
I added my analysis for sheen to the end  of my PMREM analysis colab: https://colab.research.google.com/drive/1T0D1VSyR4AllqIJTQAraEIzjlb5h4FKH?authuser=1#scrollTo=fk2vTRGzfdoN

Please excuse the state of my python; it was a bit of a rush and  I  should really brush up on numpy.

File: src/renderers/shaders/ShaderLib/meshphysical.glsl.js [hunk 1]
```diff
@@ -185,6 +185,14 @@ void main() {
 
 	#endif
 
+	#ifdef USE_SHEEN
+
+		float sheen = max( material.sheenColor.r, max( material.sheenColor.g, material.sheenColor.b ) );
```
Comment (WestLangley) [inline src/renderers/shaders/ShaderLib/meshphysical.glsl.js:190]:
```glsl
float sheen = max3( material.sheenColor );
```

File: src/renderers/shaders/ShaderChunk/lights_physical_pars_fragment.glsl.js [hunk 2]
```diff
@@ -22,6 +22,25 @@ struct PhysicalMaterial {
 
 // temporary
 vec3 clearcoatSpecular = vec3( 0.0 );
+vec3 sheenSpecular = vec3( 0.0 );
+
+// This is a curve-fit approxmation to the "Charlie sheen" BRDF integrated over the hemisphere from 
+// Estevez and Kulla 2017, "Production Friendly Microfacet Sheen BRDF"
+float IBLSheenBRDF( const in vec3 normal, const in vec3 viewDir, const in float r) {
```
Comment (WestLangley) [inline src/renderers/shaders/ShaderChunk/lights_physical_pars_fragment.glsl.js:29]:
```glsl
const in float roughness
```

**Comment by WestLangley** [comment]
>I did my best to follow the style of the ClearCoat extension.

That was the correct thing to do.

>I have an energy preservation term in there too that also came from my curve fitting.

Since you are implementing your own sheen model and sheen energy preservation term -- something different than Dassault, glTF, or Filament -- this should be documented, and a link provided in the source code. This goes for PMREM, too.

>The sheen layer is the top layer, above clear coat.

Actually, glTF spec calls for the sheen later to be under the clear coat layer. I think we should be glTF compliant.

**Comment by elalish** [comment]
> Actually, glTF spec calls for the sheen later to be under the clear coat layer. I think we should be glTF compliant.

News to me! Honestly that  sounds like a  spec error to me, but I'll follow up in the next WG call. The difference will be hard to notice in practice, but I'll fix it if it turns out I'm wrong. 

I'll add a link to my write-up.

**Comment by mrdoob** [comment]
Thanks!

**Comment by WestLangley** [comment]
>outgoingLight = outgoingLight * ( 1.0 - 0.157 * sheen ) + sheenSpecular;

The 0.157 magic number has to be referenced with a link, too.

And for clarity, something like:

```glsl
float sheenEnergyComp = 1.0 - 0.157 * max3( material.sheenColor );

outgoingLight = outgoingLight * sheenEnergyComp + sheenSpecular;
```

**Comment by WestLangley** [comment]
@mrdoob Can't you allow the review process to complete? What's the deal? Grrr...

**Comment by mrdoob** [comment]
Ah, sorry! I'll wait until your approval next time 😅

**Comment by mrdoob** [comment]
@WestLangley https://github.com/mrdoob/three.js/commit/820a605fdfd4967d2c2ce88af680e7f273886f2d

**Comment by WestLangley** [comment]
@mrdoob As I [said](https://github.com/mrdoob/three.js/pull/23069#issuecomment-999979741), a reference link is required -- perhaps the same one, or at least some explanation of where the magic number comes from.

This code cannot be _maintained by others_ if the physical modeling methods are not documented.

/ping @elalish

**Pull request merged**